packages/web-worker/src/notifier.js

```diff
@@ -0,0 +1,62 @@
+
+import Client from '@bugsnag/core/client'
+import { schema as coreSchema } from '@bugsnag/core/config'
+import delivery from '@bugsnag/delivery-fetch'
+import pluginClientIp from '@bugsnag/plugin-client-ip'
+import pluginWindowOnError from '@bugsnag/plugin-window-onerror'
+import config from './config'
+import pluginBrowserDevice from '@bugsnag/plugin-browser-device'
+import pluginPreventDiscard from './prevent-discard'
+
+const name = 'Bugsnag Web Worker'
+const url = 'https://github.com/bugsnag/bugsnag-js'
+const version = __VERSION__ // eslint-disable-line no-undef
+
+// extend the base config schema with some worker-specific options
+const schema = { ...coreSchema, ...config }
+
+export const Bugsnag = {
+  createClient: (opts) => {
+    // handle very simple use case where user supplies just the api key as a string
+    if (typeof opts === 'string') opts = { apiKey: opts }
+    if (!opts) opts = {}
+
+    // eslint-disable-next-line no-undef
+    const internalPlugins = [pluginClientIp, pluginBrowserDevice(navigator, null), pluginPreventDiscard, pluginWindowOnError(self)]
```

Comment:
`plugin-window-onerror` creates events with a component of `'window onerror'` - should we pass in `'worker onerror'` or similar?

Reply:
Yeah, or maybe `self onerror` since that's the object with the `onerror` event? I'm not sure if that's more confusing than `worker onerror` though

Reply:
@gareththackeray, @tomlongridge - thoughts on the above?

Reply:
For context, this is only used as the name of the metadata tab when you pass a non-error to `notify` (it's the value of `component` in this snippet):

https://github.com/bugsnag/bugsnag-js/blob/aced3db494a4a779cb2911dbfbdd9e6c529c807c/packages/core/event.js#L228-L234

Reply:
`worker onerror` works for me